### ISSUE TITLE
Starting the node providing a --config file (and nothing else) ignores the datadir explicitly given there.

### ISSUE THREAD

**Issue by cfc-ctf** [paradigmxyz/reth#6425]
### Describe the bug

Starting the node providing a `--config` file (and nothing else) ignores the `datadir` explicitly given in said file, in my case a toml file exactly as per the one that reth creates while sending in parameters at start — whether in the command line or from `reth.service`.
Instead, the hardcoded dir is used. 

May I suggest that you either:
**a)** do not hardcode any data dir, but instead force users to always provide a datadir path,
…or…
**b)** honor the --config file provided [while starting], fetching all parameters, including datadir from that provided toml file 

### Steps to reproduce

1. Starting the node providing a `--config` file (and nothing else) ignores the datadir explicitly given there.
2. at this point, reth ignores the `datadir` specified in that toml.
3. actually, I believe more parameters are ignored, because I never succeeded in properly starting reth with something like `reth node --datadir /my-data-path --config /my-config-path/my-config.toml`

### Node logs

```text
N/A
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

tried with alpha14 and alpha16

### What database version are you on?

Started in alpha14, now running alpha16

### What type of node are you running?

Archive (default)

### If you've built Reth from source, provide the full command you used

RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc,asm-keccak

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mattsse** [comment]
> Starting the node providing a --config file (and nothing else) ignores the datadir explicitly given there.

do you mean a `datadir` entry in your reth.toml?

I don't think we even write the datadir in there so, not exactly sure what you're referring to here exactly

**Comment by cfc-ctf** [comment]
Hi Matthias,

yes, i mean a datadir in the toml.

If not including a datadir in toml is intentional (by design), I still
argue that it would make sense if it were included — or, at least, honored
by the --config starting parameter.

In any case, the best scenario is to *not* include a default hard-coded
dir, because folks running nodes would most likely use a reth.service file
to run it as a daemon, and it would *never* make sense to run it under the
root user's local files!!
Especially since the datadir of reth would normally be located in a folder
of a high-IOPS disk, and not likely be located within the / disk.

Thanks in advance!

**Comment by cfc-ctf** [comment]
Any progress on this?

**Comment by Rjected** [comment]
> In any case, the best scenario is to *not* include a default hard-coded dir, because folks running nodes would most likely use a reth.service file to run it as a daemon, and it would *never* make sense to run it under the root user's local files!!

This happens in userspace often, I don't fully agree. The hard-coded dir makes running the node easy when you don't need to specify the dir, and I like the feature, especially for testnets. When a user _does_ have a separate high-iops / high capacity disk, it's definitely their responsibility to specify the flag, read the command `--help` entry, and understand how to use reth.

IMO it would be productive to focus on whether or not we should let users configure their datadir from the `config.toml`. I can see it being a useful feature.

**Comment by cfc-ctf** [comment]
Fair enough, allowing users to configure their `datadir` from the `config.toml` would be the cleanest path without disrupting current init assumptions.

**Comment by cfc-ctf** [comment]
Will that arrive in the Beta version?

**Comment by winksaville** [comment]
As a noob, I was looking for this feature, where the `datadir` can be configured via a "config file". But as pointed out that's not yet supported, but I did an experiment and if you pass `--datadir <DATA_DIR>` than the location of the default config file is `<DATA_DIR>/reth.toml` so I you can control the location of the config file using either `--datadir` or `--config`.
```
$ reth node --datadir /home/kendall/eth2-data/reth/holesky --help
Start the node

Usage: reth node [OPTIONS]

Options:
      --datadir <DATA_DIR>
          The path to the data dir for all reth files and subdirectories.
          
          Defaults to the OS-specific data directory:
          
          - Linux: `$XDG_DATA_HOME/reth/` or `$HOME/.local/share/reth/`
          - Windows: `{FOLDERID_RoamingAppData}/reth/`
          - macOS: `$HOME/Library/Application Support/reth/`
          
          [default: default]

      --config <FILE>
          The path to the configuration file to use.
...
      --log.file.directory <PATH>
          The path to put log files in
          
          [default: /home/kendall/.cache/reth/logs]
...
```

Something that is also confusing is the help says "The path to the data dir for all reth files and subdirectories.". To my mind "all reth files" would also include the location of things like the logs. But that wasn't the case, when I specified `--datadir /home/kendall/eth2-data/reth/holesky` I expected to see the `--log.file.directory` to be `/home/kendall/eth2-data/reth/holesky` but no, on my linux machine it is the value shown above, `/home/kendall/.cache/reth/logs`.

Actually, there seems to be several places `<PATH>` might be preferable and I'd suggest that all the "help" use `<PATH>` where appropriate; such as `--config <PATH>` , `--datadir <PATH>`, `--peers-file <PATH>` as it is already used in 5 other places in the `--help`:
```
$ reth node --help | rg '<PATH>'
      --trusted-setup-file <PATH>
      --p2p-secret-key <PATH>
      --authrpc.jwtsecret <PATH>
      --debug.engine-api-store <PATH>
      --log.file.directory <PATH>
```

**Comment by cfc-ctf** [comment]
I still think that forcing to enter a single path for everything related to
reth would be a great start. Of course, we could use more/narrowed paths,
but having to provide at least one mandatory path will eliminate issues for
users.

**Comment by winksaville** [comment]
Personally I don't think it's necessary to force always having a single location. But I would like the option of providing a single location. Right now my reth.service ExecStart has two path entries, datadir and  log.file.directory. IMO I should only have to supply one.

And actually, as discussed, I should be able to provide my defaults for any/all parameters in a config.toml and then override those from the command line as I desire.
